### ISSUE TITLE
softmax_cross_entropy_with_logits() behaves differently with it's comment

### ISSUE THREAD

**Issue by metorm** [tensorflow/tensorflow#8414]
version: 1.0.1 CPU compiled by cmake from master branch.

According to [the document](https://www.tensorflow.org/api_docs/python/tf/nn/softmax_cross_entropy_with_logits), `softmax_cross_entropy_with_logits()` accept inputs with any shape. However, the comments in the source code says:

`logits` and `labels` must have the same shape `[batch_size, num_classes]`  and the same dtype (either `float16`, `float32`, or `float64`).

meaning that only rank-2 tensors are allowed.

I tried the following code to construct a net which performs binary-classify on each pixel of an image :

```
    final_two_channel = tf.layers.conv2d(
        inputs=res_out_iter,
        filters=2,
        kernel_size=[1, 1],
        padding='same',
        activation=tf.nn.sigmoid
    )
    # logits = tf.reshape(tensor=final_two_channel, shape=[-1, 2], name='flatten_net_out')
    label_one_hot = tf.one_hot(
        indices=output_field,
        depth=2,
        on_value=1, off_value=0,
        name='label_one_hot'
    )
    crs_etp_loss = tf.nn.softmax_cross_entropy_with_logits(labels=label_one_hot, logits=final_two_channel)
    print(crs_etp_loss.shape)
    crs_etp_loss = tf.reshape(tensor=crs_etp_loss, shape=[-1, field_width * field_width])
    crs_etp_loss = tf.reduce_mean(input_tensor=crs_etp_loss, axis=1)

    optimizer = tf.train.AdamOptimizer(learning_rate=0.001).minimize(loss=crs_etp_loss)
```

The program runs without error and `print(crs_etp_loss.shape)` gives `(?, 512, 128)`, but I am not sure if it's safe/correct to do like this.

I posted a question on [stackoverflow](http://stackoverflow.com/questions/42799167/is-it-safe-to-use-softmax-cross-entropy-with-logits-when-each-instance-has-mul) about this.

**Comment by skye** [comment]
I'm gonna close this issue as it's not a bug or feature request. Thanks for posting to StackOverflow, that's the preferred forum for usage questions like this.
